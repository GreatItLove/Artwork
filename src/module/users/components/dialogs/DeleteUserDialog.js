import React from 'react';
import { Dialog } from 'material-ui';
import PropTypes from 'prop-types';

import Panel from '../../../common/components/Panel';
import CloseIcon from '../../../common/components/header/CloseIcon';
import SubmitBtnLoader from '../../../common/components/form/SubmitBtnLoader';

const propTypes = {
  openDialog: PropTypes.bool.isRequired,
  loading: PropTypes.bool.isRequired,
  userId: PropTypes.oneOfType([PropTypes.bool, PropTypes.number]).isRequired,
  deleteUsers: PropTypes.func.isRequired,
  closeDialog: PropTypes.func.isRequired,
  muiTheme: PropTypes.shape({
    palette: PropTypes.shape({
      colorBrandOrange: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

function DeleteUserDialog({
  closeDialog,
  deleteUsers,
  userId,
  openDialog,
  muiTheme,
  ...rest
}) {
  const customContentStyle = {
    width: '57%',
    position: 'fixed',
    top: '0',
    bottom: '0',
    left: '0',
    right: '0',
    maxWidth: 'none',
  };
  const actions = (
    <div className="col-sm-12">
      <div className="row">
        <div className="col-md-7" />
        <div className="col-md-2 my-auto">
          <span className="users cancel-link" onClick={closeDialog}>
            CANCEL
          </span>
        </div>
        <div className="col-md-3">
          <SubmitBtnLoader
            backgroundColor={muiTheme.palette.colorBrandOrange}
            labelColor="#ffffff"
            primary={false}
            style={{ boxShadow: 'none' }}
            label="Delete User"
            onClick={() => {
              deleteUsers(userId);
            }}
            {...rest}
          />
        </div>
      </div>
    </div>
  );
  return (
    <Dialog
      open={openDialog}
      bodyStyle={{ overflowY: 'unset' }}
      contentStyle={customContentStyle}
      overlayStyle={{ backgroundColor: '#797979', opacity: '0.8' }}
      paperProps={{ style: { backgroundColor: 'none' }, zDepth: 0 }}
      onRequestClose={closeDialog}>
      <Panel
        title="Delete user"
        panelHeadingClassName="users-from-panel-heading"
        rightElement={<CloseIcon closeDialog={closeDialog} />}
        actions={actions}>
        Do you really want to delete the user?
      </Panel>
    </Dialog>
  );
}

DeleteUserDialog.propTypes = propTypes;

export default DeleteUserDialog;
