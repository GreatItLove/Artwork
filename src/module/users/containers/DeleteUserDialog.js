import { connect } from 'react-redux';
import { compose } from 'recompose';
import muiThemeable from 'material-ui/styles/muiThemeable';

import { deleteUserRequest, closeUserDeleteDialog } from '../usersActions';
import LoaderHOC from '../../common/HOCs/LoaderHOC';
import DeleteUserDialog from '../components/dialogs/DeleteUserDialog';

const mapStateToDispatch = dispatch => ({
  deleteUsers: userId => {
    dispatch(deleteUserRequest(userId));
  },
  closeDialog: () => {
    dispatch(closeUserDeleteDialog());
  }
});

const mapStateToProps = state => ({
  openDialog: state.users.deleting !== false,
  loading: state.users.deleting === true,
  userId: state.users.deleting
});

const enhance = compose(connect(mapStateToProps, mapStateToDispatch), LoaderHOC('loading', true), muiThemeable());
export default enhance(DeleteUserDialog);
