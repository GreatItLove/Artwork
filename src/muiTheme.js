import getMuiTheme from 'material-ui/styles/getMuiTheme';

const muiTheme = getMuiTheme({
  palette: {
    primary1Color: '#258df2',
    accent1Color: '#40c741',
    colorBrandBlue: '#1e6291',
    colorBrandGrey: '#e7e4d8',
    colorBrandOrange: '#e18927'
  }
});

export default muiTheme;
