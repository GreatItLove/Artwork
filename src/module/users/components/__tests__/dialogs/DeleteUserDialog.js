import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import DeleteUserDialog from '../../dialogs/DeleteUserDialog';
import muiTheme from '../../../../../../muiTheme';

describe('DeleteUserDialog', () => {
  it('Snapshot of DeleteUserDialog', () => {
    const props = {
      openDialog: true,
      loading: false,
      userId: 6546,
      deleteUsers: jest.fn(),
      closeDialog: jest.fn(),
      muiTheme: muiTheme,
    };
    const wrapper = shallow(<DeleteUserDialog {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
