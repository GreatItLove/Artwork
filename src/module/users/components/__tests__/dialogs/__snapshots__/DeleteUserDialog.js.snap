// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteUserDialog Snapshot of DeleteUserDialog 1`] = `
<Dialog
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  bodyStyle={
    Object {
      "overflowY": "unset",
    }
  }
  contentStyle={
    Object {
      "bottom": "0",
      "left": "0",
      "maxWidth": "none",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "width": "57%",
    }
  }
  modal={false}
  onRequestClose={[Function]}
  open={true}
  overlayStyle={
    Object {
      "backgroundColor": "#797979",
      "opacity": "0.8",
    }
  }
  paperProps={
    Object {
      "style": Object {
        "backgroundColor": "none",
      },
      "zDepth": 0,
    }
  }
  repositionOnUpdate={true}
>
  <Component
    actions={
      <div
        className="col-sm-12"
      >
        <div
          className="row"
        >
          <div
            className="col-md-7"
          />
          <div
            className="col-md-2 my-auto"
          >
            <span
              className="users cancel-link"
              onClick={[Function]}
            >
              CANCEL
            </span>
          </div>
          <div
            className="col-md-3"
          >
            <SubmitBtnLoader
              backgroundColor="#e18927"
              label="Delete User"
              labelColor="#ffffff"
              loading={false}
              onClick={[Function]}
              primary={false}
              style={
                Object {
                  "boxShadow": "none",
                }
              }
            />
          </div>
        </div>
      </div>
    }
    panelHeadingClassName="users-from-panel-heading"
    rightElement={
      <CloseIcon
        closeDialog={[Function]}
      />
    }
    title="Delete user"
  >
    Do you really want to delete the user?
  </Component>
</Dialog>
`;
